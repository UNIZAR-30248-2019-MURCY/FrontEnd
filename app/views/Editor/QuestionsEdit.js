import React, {Component} from 'react';
import {
    View,
    StyleSheet,
    FlatList,
    ScrollView,
    SafeAreaView
} from 'react-native'
import {Button, Text, ListItem} from 'react-native-elements';
import ActionButton from 'react-native-action-button';
import Icon from 'react-native-vector-icons/FontAwesome';
import {retrieveItem} from "../../modules/AsyncStorage/retrieve";
import {listQuestions } from "../../services/quiz/questionFuncs";


export default class QuestionsEdit extends Component {

    constructor(props) {
        super(props);
        this.state = {
            token: '',
            error: '',
            data: [
                {
                    id: 1,
                    title: '¿Cual es la capital de España?',
                    description: 'Hey q pasa',
                    options: [
                        {
                            title: 'Madrid',
                            correct: true
                        },
                        {
                            title: 'Zgz',
                            correct: false
                        },

                    ]
                },
                {
                    id: 2,
                    title: '¿Cual es la capital de Aragón?',
                    description: 'Hey q pasa',
                    options: [
                        {
                            title: 'Huesca',
                            correct: false
                        },
                        {
                            title: 'Zgz',
                            correct: true
                        },
                        {
                            title: 'Teruel',
                            correct: false
                        },
                    ]
                }
            ]
        }
    }

    
    componentDidMount() {
        retrieveItem('token')
            .then(data => {
                this.setState({token: data})
                listQuestions(this.state.token)
                    .then((data) => {
                        console.log(data);
                        this.setState(data);
                    })
                    .catch((error) => {
                        this.setState( {error: error.message})
                    })
            })
    }


    render() {
        return (

            <View style={styles.container}>
                <View style={styles.containerTitle}>
                    <Text h2>Questions</Text>
                </View>
                <ScrollView>
                    <SafeAreaView style={styles.containerQuestions}>
                        <FlatList
                            data={this.state.data}
                            keyExtractor={item => item.id.toString()}
                            renderItem={({item}) => (
                                <ListItem
                                    title={item.title}
                                    rightIcon={
                                        <Icon
                                            name="chevron-right"
                                            size={12}
                                            color="grey"
                                        />
                                    }
                                    bottomDivider
                                    onPress={() => {
                                        this.props.navigation.navigate('EditRemoveQuestion', { info: item });
                                    }}
                                />
                            )}
                        />

                    </SafeAreaView>
                </ScrollView>
                <ActionButton
                    className='create-button'
                    hideShadow={true}
                    buttonColor="grey"
                    onPress={() => {
                        this.props.navigation.navigate('CreateQuestion');
                    }}
                />

            </View>


        )
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        //justifyContent: 'center',
        padding: 20
    },
    containerTitle: {
        alignItems: 'center',
        marginTop: 50,
        padding: 20
    },
    containerQuestions: {
        flex: 1,
        marginTop: 20,
    },
    button: {
        width: 150,
        height: 55,
        marginTop: 20,
        margin: 10,
        backgroundColor: 'grey',
        borderRadius: 14
    },

})
