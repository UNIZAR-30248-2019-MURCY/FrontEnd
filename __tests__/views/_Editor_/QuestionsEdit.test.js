import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import QuestionsEdit from "../../../app/views/Editor/QuestionsEdit";

describe('<QuestionsEdit />', () => {
    it('QuestionsEdit renders without crashing', () => {
        const rendered = renderer.create(<QuestionsEdit />).toJSON();
        expect(rendered).toBeTruthy();
    });
    it('QuestionsEdit test against snapshot', () => {
        const welcomePage = renderer.create(<QuestionsEdit />).toJSON();
        expect(welcomePage).toMatchSnapshot();
    });
    it('Button Create Question', () => {
        const navigationMock = { navigate: jest.fn() };
        const wrapper = shallow(<QuestionsEdit navigation={navigationMock} />);
        wrapper.find('.create-button').simulate('press');
        expect(navigationMock.navigate.mock.calls.length).toEqual(1);
        expect(navigationMock.navigate.mock.calls[0]).toEqual(['CreateQuestion']);
    });
});
