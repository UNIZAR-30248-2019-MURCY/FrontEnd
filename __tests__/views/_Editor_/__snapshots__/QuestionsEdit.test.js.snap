// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QuestionsEdit /> .renderItem should match the snapshot 1`] = `ShallowWrapper {}`;

exports[`<QuestionsEdit /> QuestionsEdit test against snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "padding": 20,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginTop": 50,
        "padding": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 42.5,
        }
      }
      theme={
        Object {
          "colors": Object {
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": Object {
              "android": Object {
                "error": "#f44336",
                "primary": "#2196f3",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "ios": Object {
                "error": "#ff3b30",
                "primary": "#007aff",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#8F0CE8",
            "success": "#52c41a",
            "warning": "#faad14",
          },
        }
      }
    >
      Questions
    </Text>
  </View>
  <RCTScrollView>
    <View>
      <RCTSafeAreaView
        emulateUnlessSupported={true}
        style={
          Object {
            "flex": 1,
            "marginTop": 20,
          }
        }
      >
        <View
          className="loadingShow"
          style={
            Array [
              undefined,
            ]
          }
        >
          <ActivityIndicator
            animating={true}
            color="grey"
            hidesWhenStopped={true}
            size="large"
          />
        </View>
        <RCTScrollView
          className="list"
          data={Array []}
          disableVirtualization={false}
          getItem={[Function]}
          getItemCount={[Function]}
          horizontal={false}
          initialNumToRender={10}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          numColumns={1}
          onContentSizeChange={[Function]}
          onEndReachedThreshold={2}
          onLayout={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={50}
          stickyHeaderIndices={Array []}
          updateCellsBatchingPeriod={50}
          viewabilityConfigCallbackPairs={Array []}
          windowSize={21}
        >
          <View />
        </RCTScrollView>
      </RCTSafeAreaView>
    </View>
  </RCTScrollView>
  <View
    pointerEvents="box-none"
    style={
      Array [
        Array [
          Object {
            "backgroundColor": "transparent",
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "elevation": undefined,
            "justifyContent": "flex-end",
            "zIndex": undefined,
          },
        ],
        undefined,
      ]
    }
  >
    <View
      pointerEvents="none"
      style={
        Object {
          "backgroundColor": "transparent",
          "bottom": 0,
          "elevation": undefined,
          "justifyContent": "flex-end",
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": undefined,
        }
      }
    />
    <View
      pointerEvents="box-none"
      style={
        Array [
          Array [
            Object {
              "backgroundColor": "transparent",
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "elevation": undefined,
              "justifyContent": "flex-end",
              "zIndex": undefined,
            },
          ],
          Object {
            "alignItems": "flex-end",
          },
          Object {
            "paddingVertical": 30,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "marginHorizontal": 30,
              "zIndex": undefined,
            },
            false,
            false,
          ]
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgba(128, 128, 128, 1)",
                "borderRadius": 28,
                "height": 56,
                "width": 56,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderRadius": 28,
                  "height": 56,
                  "justifyContent": "center",
                  "transform": Array [
                    Object {
                      "scale": 1,
                    },
                    Object {
                      "rotate": "0deg",
                    },
                  ],
                  "width": 56,
                }
              }
            >
              <Text
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "color": "rgba(255, 255, 255, 1)",
                    "fontSize": 24,
                    "marginTop": -4,
                  }
                }
              >
                +
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
